packages/spacecat-shared-content-client/src/clients/content-client.js

```diff
@@ -97,6 +98,68 @@ const validateMetadata = (metadata) => {
   }
 };
 
+const validateRedirects = (redirects) => {
+  if (!Array.isArray(redirects)) {
+    throw new Error('Redirects must be an array');
+  }
+
+  if (!redirects.length) {
+    throw new Error('Redirects must not be empty');
+  }
+
+  for (const redirect of redirects) {
+    if (!isObject(redirect)) {
+      throw new Error('Redirect must be an object');
+    }
+
+    if (!hasText(redirect.from)) {
+      throw new Error('Redirect must have a valid from path');
+    }
+
+    if (!hasText(redirect.to)) {
+      throw new Error('Redirect must have a valid to path');
+    }
+
+    if (!redirect.from.startsWith('/')) {
+      throw new Error('Redirect from path must start with a slash');
+    }
+
+    if (!redirect.to.startsWith('/')) {
+      throw new Error('Redirect to path must start with a slash');
+    }
+
+    if (redirect.from === redirect.to) {
+      throw new Error('Redirect from and to paths must be different');
+    }
+  }
+};
+
+const removeDuplicatedRedirects = (currentRedirects, newRedirects) => {
+  const redirectsSet = new Set();
+  currentRedirects.map(({ from, to }) => `${from}:${to}`)
+    .forEach((redirectRule) => redirectsSet.add(redirectRule));
+
+  const newRedirectsClean = [];
+  newRedirects.forEach((redirectRule) => {
+    const { from, to } = redirectRule;
+    const strRedirectRule = `${from}:${to}`;
+    if (!redirectsSet.has(strRedirectRule)) {
+      redirectsSet.add(strRedirectRule);
+      newRedirectsClean.push(redirectRule);
+    } // TODO: report duplicated entry??
+  });
+  return newRedirectsClean;
+};
+
+const detectRedirectLoops = (currentRedirects, newRedirects) => {
+  const redirectsGraph = new Graph();
+  [...currentRedirects, ...newRedirects].forEach((r) => redirectsGraph.addEdge(r.from, r.to));
+  if (hasCycle(redirectsGraph)) {
+    // TODO: report cycle path??
+    throw new Error('Redirect cycle detected');
```

Comment:
In this case I see we are not adding any of the entries. Could we check after each edge if there is a cycle and just not add that specific edge ?